### PR TITLE
Persistence code-style refactoring

### PR DESCRIPTION
**Pull request by alexvaluyskiy**
Added XML comments
Eventsoured is not a partial class anymore
Moved messages to `*.Messages.cs` files
Changed `readonly fields` to `auto get properties`
Updated the codebase to C#6

### CODE REVIEW

**Comment by Horusiath** [comment]
Could you roughly describe your changes? Also I think, that maybe `IMessage` is not the most descriptive name. Maybe we should call it `IPersisted`, `IPersistent` or `IPersistentMessage`?

File: src/core/Akka.Persistence/Eventsourced.Messages.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        public Action<object> Handler { get; }
+    }
+
+    /// <summary>
+    /// Unlike <see cref="StashingHandlerInvocation"/> this one does not force actor to stash commands.
+    /// Originates from <see cref="Eventsourced.PersistAsync{TEvent}(TEvent,System.Action{TEvent})"/> 
+    /// or <see cref="Eventsourced.DeferAsync{TEvent}"/> method calls.
+    /// </summary>
+    internal sealed class AsyncHandlerInvocation : IPendingHandlerInvocation
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Eventsourced.Messages.cs]: Changed to `internal`

File: src/core/Akka.Persistence/Persistence.cs [hunk 2]
```diff
@@ -535,20 +547,4 @@ public interface IPersistenceStash : IWithUnboundedStash
         /// </summary>
         IStashOverflowStrategy InternalStashOverflowStrategy { get; }
     }
-
-    /// <summary>
-    /// TBD
-    /// </summary>
-    public interface IJournalPlugin
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Persistence.cs]: This interface is not used anywhere

File: src/core/Akka.Persistence/Eventsourced.Messages.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using Akka.Actor;
+
+namespace Akka.Persistence
+{
+    internal interface IPendingHandlerInvocation
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Eventsourced.Messages.cs]: Changed to internal

File: src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt [hunk 4]
```diff
@@ -1197,399 +1150,4 @@ namespace Akka.Persistence.Snapshot
         protected abstract System.Threading.Tasks.Task SaveAsync(Akka.Persistence.SnapshotMetadata metadata, object snapshot);
     }
 }
-public sealed class AtLeastOnceDeliverySnapshot : Google.ProtocolBuffers.GeneratedMessage<AtLeastOnceDeliverySnapshot, AtLeastOnceDeliverySnapshot.Builder>
```
Comment (alexvaluyskiy) [inline src/core/Akka.API.Tests/CoreAPISpec.ApprovePersistence.approved.txt]: Protobuf messages should not be exposed publically

File: src/core/Akka.Persistence/InternalExtensions.cs [hunk 5]
```diff
@@ -1,34 +0,0 @@
...
-    /// </summary>
-    internal static class InternalExtensions
-    {
-        /// <summary>
-        /// Enqueues provided <paramref name="message"/> at the beginning of the internal actor cell mailbox message queue.
-        /// Requires current actor to use unbounded deque-based message queue. It will fail otherwise.
-        /// </summary>
-        /// <param name="context">TBD</param>
-        /// <param name="message">TBD</param>
-        public static void EnqueueMessageFirst(this IActorContext context, object message)
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/InternalExtensions.cs]: This method is not used anywehere

File: src/core/Akka.Persistence/JournalProtocol.cs [hunk 6]
```diff
@@ -285,34 +288,14 @@ public override string ToString()
...
-        /// TBD
+        /// The singleton instance of <see cref="WriteMessagesSuccessful"/>.
         /// </summary>
-        public static readonly WriteMessagesSuccessful Instance = new WriteMessagesSuccessful();
+        public static WriteMessagesSuccessful Instance { get; } = new WriteMessagesSuccessful();
 
         private WriteMessagesSuccessful() { }
-
-        /// <inheritdoc/>
-        public bool Equals(WriteMessagesSuccessful other)
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/JournalProtocol.cs]: Singleton classes should be checked only by type equality

File: src/core/Akka.Persistence/Persistence.cs [hunk 7]
```diff
@@ -17,36 +17,20 @@
 
 namespace Akka.Persistence
 {
-    /// <summary>
-    /// TBD
-    /// </summary>
-    internal struct PluginHolder
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Persistence.cs]: Why it was `struct`?

File: src/core/Akka.Persistence/Persistent.cs [hunk 8]
```diff
@@ -15,50 +15,14 @@
 namespace Akka.Persistence
 {
     /// <summary>
-    /// TBD
-    /// </summary>
-    public interface IPersistentIdentity
```
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Persistent.cs]: Moved to another file

**Comment by alexvaluyskiy** [comment]
@Horusiath This pull request mostly about XML documentation and updating to C#6. Also I've created a couple of `*.Messages.cs` files and moved the messages there.
I've commented the most notable changes

**Comment by Aaronontheweb** [comment]
API spec failed.

**Comment by alexvaluyskiy** [comment]
@Aaronontheweb I will update the API spec, when someone has finished the review

File: src/core/Akka.Persistence/Journal/Tagged.cs [hunk 9]
```diff
@@ -19,20 +19,11 @@ namespace Akka.Persistence.Journal
     /// journal implementation for more information.
     /// The journal will unwrap the event and store the <see cref="Payload"/>.
     /// </summary>
-    public struct Tagged
```
Comment (Danthar) [inline src/core/Akka.Persistence/Journal/Tagged.cs]: Why did you change this? Why was it a struct in the first place ?

File: src/core/Akka.Persistence/Persistent.cs [hunk 10]
```diff
@@ -145,13 +96,15 @@ public AtomicWrite(IImmutableList<IPersistentRepresentation> payload)
...
-        /// TBD
+        /// This persistent message's payload.
         /// </summary>
         public object Payload { get; private set; }
+
         /// <summary>
         /// TBD
         /// </summary>
-        public IActorRef Sender { get; private set; }
+        public IActorRef Sender { get; }
```
Comment (Danthar) [inline src/core/Akka.Persistence/Persistent.cs]: Considering you added XML comments for lots of other part. You might as well do these as well ;)

File: src/core/Akka.Persistence/PersistentActor.Messages.cs [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+    /// 
+    /// If recovery starts from a snapshot, the <see cref="PersistentActor"/> is offered that snapshot with a
+    /// <see cref="SnapshotOffer"/> message, followed by replayed messages, if any, that are younger than the snapshot, up to the
+    /// specified upper sequence number bound (<see cref="ToSequenceNr"/>).
+    /// </summary>
+    [Serializable]
+    public sealed class Recovery
+    {
+        /// <summary>
+        /// TBD
```
Comment (Danthar) [inline src/core/Akka.Persistence/PersistentActor.Messages.cs]: This comment as well.

**Comment by Danthar** [comment]
Apart for some xml docs which you skipped :P. And the change of some types from struct to class.  
LGTM.
Not sure why those types were struct to begin with. At first glance it seems like they are not types used in hot paths. So any advantages in memory usage seems negligible ?

File: src/core/Akka.Persistence/Journal/Tagged.cs (hunk 9 above)
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Journal/Tagged.cs]: Have no idea why it was a struct @Horusiath

File: src/core/Akka.Persistence/Persistent.cs (hunk 10 above)
Comment (alexvaluyskiy) [inline src/core/Akka.Persistence/Persistent.cs]: Ok
